MonMaraicher/Features/FarmerDetail/FarmerDetailsViewModelTests.swift

```diff
@@ -2,65 +2,58 @@
 //  FarmerDetailsViewModelTests.swift
 //  MonMaraicherTests
 //
-//  Created by Yves Charpentier on 11/03/2024.
+//  Created by Yves Charpentier on 03/05/2024.
 //
 
 import XCTest
+import CoreLocation
 @testable import MonMaraicher
 
 final class FarmerDetailsViewModelTests: XCTestCase {
 
-    func testFarmerCityShouldReturnCityCapitalized() {
-        // Given
-        let farmerMock = Farmer.makeMock(city: "paris")
-        let farmerDetail = FarmerDetailsViewModel(farmer: farmerMock)
+    let farmerServiceMock = FarmerServiceMock()
 
-        // When
-        let expectedCity = "Paris"
+    func testFarmerNameShouldReturnCorrectNameOfFarmer() async {
+        do {
+            // Given
+            let farmers = try await farmerServiceMock.searchFarmers(location: CLLocation(latitude: 34, longitude: 34))
+            let farmer = farmers.items.first!
 
-        // Then
-        XCTAssertEqual(farmerDetail.city, expectedCity)
-    }
-
-    func testFarmerAddressShouldReturnFormattedAddressWithoutStreetNumber() {
-        // Given
-        let farmerMock = Farmer.makeMock(streetName: "rue de la paix", zipCode: 75000, city: "paris")
-        let farmerDetail = FarmerDetailsViewModel(farmer: farmerMock)
+            // When
+            let expectedFarmerName = "L'Abeille du Pic"
 
-        // When
-        let expectedFormattedAddress = "Rue De La Paix\n75000 Paris"
+            // Then
+            XCTAssertEqual(farmer.businessName, expectedFarmerName)
+        } catch {
 
-        // Then
-        XCTAssertEqual(farmerDetail.address, expectedFormattedAddress)
+        }
     }
 
-    func testFarmerAddressShouldReturnFormattedAddressWithStreetNumber() {
-        // Given
-        let farmerMock = Farmer.makeMock(streetNumber: 5, streetName: "rue de la paix", zipCode: 75000, city: "paris")
-        let farmerDetail = FarmerDetailsViewModel(farmer: farmerMock)
-
-        // When
-        let expectedFormattedAddress = "5 Rue De La Paix\n75000 Paris"
-
-        // Then
-        XCTAssertEqual(farmerDetail.address, expectedFormattedAddress)
-    }
+    func testFarmerAddressShouldReturnFormattedAddress() async {
+        do {
+            // Given
+            let farmers = try await farmerServiceMock.searchFarmers(location: CLLocation(latitude: 34, longitude: 34))
```

Comment:
You should not have to have that. Here you are testing the details viewModel and your test can fail because of a service. Your viewModel doesn't know anything about it so it doesn't make sense to use a service here

Reply:
I think I need help for unit tests